ui/pages/permissions-connect/permissions-connect.component.js

```diff
@@ -113,8 +116,10 @@ export default class PermissionConnect extends Component {
 
     if (history.location.pathname === connectPath && !isRequestingAccounts) {
       ///: BEGIN:ONLY_INCLUDE_IN(flask)
-      if (isSnap) {
+      if (isSnap && !permissionsRequest.newPermissions) {
```

Comment:
It would, but I'm not so keen on updating existing extension selectors to have snaps logic. I think it would make sense to do that once snaps hits main extension release. I've left a comment in the code to update when that happens.